mcm/model_argument_transform.py

```diff
@@ -9,21 +10,27 @@
 categorical_param_name_regex = re.compile(categorical_param_name_pattern)
 
 
-class IndicatorTransform:
+class AbstractBaseTransform(metaclass=ABCMeta):
```

Comment:
aha! i didn't know how to do abstract classes/interfaces in python...this is great.

my general instinct is, particularly for a complex codebase, is to get the typing right to try and move errors from runtime to compile time. i also, agree it seems to make sense in this instance.